diff -Naur mingw-w64-v3.1.0-orig/mingw-w64-headers/crt/float.h mingw-w64-v3.1.0/mingw-w64-headers/crt/float.h
--- mingw-w64-v3.1.0-orig/mingw-w64-headers/crt/float.h	2012-07-04 06:06:41.000000000 -0700
+++ mingw-w64-v3.1.0/mingw-w64-headers/crt/float.h	2015-07-07 10:57:36.181290307 -0700
@@ -59,6 +59,15 @@
 	#define DBL_MIN		__DBL_MIN__
 	#define LDBL_MIN	__LDBL_MIN__
 	
+	/* The difference between 1 and the least value greater than 1 that is
+	 *    representable in the given floating point type, b**1-p.  */
+	#undef FLT_EPSILON
+	#undef DBL_EPSILON
+	#undef LDBL_EPSILON
+	#define FLT_EPSILON __FLT_EPSILON__
+	#define DBL_EPSILON __DBL_EPSILON__
+	#define LDBL_EPSILON    __LDBL_EPSILON__
+
 	/* Needed for libjava building - Victor K. */
 
 	/* Radix of exponent representation, b. */

